packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,581 @@
+  /// that uniquely identifies this widget. This key is what will be returned
+  /// in the [onValueChanged] callback when a new value from the [children] map
+  /// is selected. Most widgets that use a segmented control will listen for
+  /// the [onValueChanged] callback and rebuild the segmented control with
+  /// the returned key as the new [groupValue] to update the currently selected
+  /// option.
+  ///
+  /// The [groupValue] must be one of the keys in the [children] map.
+  /// The [groupValue] is the currently selected value for the segmented control.
+  /// If no [groupValue] is provided, or the [groupValue] is null, no widget will
+  /// appear as selected.
+  SegmentedControl({
+    Key key,
+    @required this.children,
+    @required this.onValueChanged,
+    this.groupValue,
+  })  : assert(children != null),
+        assert(children.length >= 2),
+        assert(onValueChanged != null),
+        assert(groupValue == null || children.keys.any((T child) => child == groupValue)),
+        // in case the default implementation of Map changes
+        assert(children is LinkedHashMap),
+        super(key: key);
+
+  /// The identifying keys and corresponding widget values in the
+  /// segmented control.
+  ///
+  /// This attribute must not be null. The length of the list must be greater
+  /// than or equal to two.
+  final Map<T, Widget> children;
+
+  /// The identifier of the widget that is currently selected.
+  ///
+  /// This must be one of the keys in the [LinkedHashMap] of [children].
+  /// If this attribute is null, no widget will be initially selected.
+  final T groupValue;
+
+  /// The callback that is called when a new option is tapped.
+  ///
+  /// This attribute must not be null.
+  ///
+  /// The segmented control passes the newly selected widget's associated key
+  /// to the callback but does not actually change state until the parent
+  /// widget rebuilds the segmented control with the new [groupValue].
+  ///
+  /// The callback provided to [onValueChanged] should update the state of
+  /// the parent [StatefulWidget] using the [State.setState] method, so that
+  /// the parent gets rebuilt; for example:
+  ///
+  /// ## Sample code
+  ///
+  /// ```dart
+  /// class SegmentedControlExample extends StatefulWidget {
+  ///   @override
+  ///   State createState() => new SegmentedControlExampleState();
+  /// }
+  ///
+  /// class SegmentedControlExampleState extends State<SegmentedControlExample> {
+  ///   final Map<int, Widget> children = <int, Widget>{};
+  ///
+  ///   int currentValue;
+  ///
+  ///   @override
+  ///   void initState() {
+  ///     super.initState();
+  ///     children[0] = const Text('Child 1');
+  ///     children[1] = const Text('Child 2');
+  ///   }
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new Container(
+  ///       child:  new SegmentedControl<int>(
+  ///         children: children,
```

Comment:
a better way to do this would be to have `children` just be set to a `const { 0: const Text('Child 1'), 1: const Text('Child 2') }`